### PR TITLE
fix: fixes multi-platform push

### PR DESCRIPTION
**Pull request by bruce-szalwinski-he**
Addresses #2.  

Thanks for the awesome plugin.  I fixed the push issue by joining the platforms.  While I was here, I added a few things:

- unit tests
- pre-commit
- bumped a few dependencies to get tests and typing working

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (bruce-szalwinski-he) [inline .pre-commit-config.yaml:1]: do some checks before committing.

File: Makefile
Comment (bruce-szalwinski-he) [inline Makefile:29]: check format and run tests

File: poetry_docker_plugin/docker_builder.py
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:286]: created a class to build the command to make it easy to test.  didn't want to mock `subprocess.run`.

File: poetry_docker_plugin/docker_builder.py
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:298]: same thing.  created a class to create the push command to make it easy to test the command was being created.

File: poetry_docker_plugin/docker_builder.py
Comment (bruce-szalwinski-he) [inline poetry_docker_plugin/docker_builder.py:401]: This is the key line of the entire PR :).

File: pyproject.toml
Comment (vagmcs) [inline pyproject.toml:111]: This should be removed.

**Review by vagmcs — COMMENTED** [review]
Thank you for fixing the issue. I just added a couple of minor comments regarding the Makefile.

File: Makefile
Comment (bruce-szalwinski-he) [inline Makefile:29]: added the `--diff` and `--no-fix` with the intention of using pre-commit to fix before commit and the github workflow to be a quality gate.

**Comment by bruce-szalwinski-he** [comment]
system tested with 

```
 poetry self add poetry-docker-plugin git+https://github.com/bruce-szalwinski-he/poetry-docker-plugin.git@multi-platform
 poetry docker --platform linux/amd64 --platform linux/arm64 --push
```

**Review by vagmcs — APPROVED** [review]
Everything seems fine. You can merge it. Please use squash and merge with a commit message starting with `bug: your message`, according to the [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0). This is helpful for me to generate changelogs for the new release from commit messages.

**Comment by bruce-szalwinski-he** [comment]
I don't have privileges to merge.  I updated the title of the PR which should become the commit message when you squash merge.  Conventional commit suggests using `fix` to patch a bug, so went with that.  Also, need to decide if you want the PR workflow (it is waiting for approval, could be updated to run on merges to main as well).  I can remove if you don't want it.

**Pull request merged**
